src/vm/gchelpers.cpp

```diff
@@ -437,7 +449,7 @@ void ThrowOutOfMemoryDimensionsExceeded()
 //
 // This is wrapper overload to handle TypeHandle arrayType
 //
-OBJECTREF AllocateArrayEx(TypeHandle arrayType, INT32 *pArgs, DWORD dwNumArgs, BOOL bAllocateInLargeHeap)
```

Comment:
It would be nice to change all these methods to just take the flags instead of multiple bools.

Reply:
It is bac to just one bool now since Uninitialized flag has effect only in the "Sz" path. So no longer a problem. 

---
In reply to: [277933949](https://github.com/dotnet/coreclr/pull/24096#discussion_r277933949) [](ancestors = 277933949)

Reply:
AllocateArrayEx does not have the problem anymore, but other method still do, e.g.:

`Alloc(size_t size, BOOL bFinalize, BOOL bContainsPointers, BOOL zeroingOptional = FALSE)`

Reply:
Ok, now taking an enum flags for everything except bAllocateInLargeHeap. That is still a bool since it does not flow as easily with other parameters and often requires to call different helpers. Flowing could be error-prone.

---
In reply to: [279240157](https://github.com/dotnet/coreclr/pull/24096#discussion_r279240157) [](ancestors = 279240157)